services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_utils.py

```diff
@@ -0,0 +1,221 @@
+import logging
+from collections import deque
+from contextlib import asynccontextmanager
+from typing import Any, AsyncIterator, Deque, Dict, Final, List, Optional, Type
+
+from fastapi import FastAPI
+from pydantic import AnyHttpUrl
+from servicelib.fastapi.long_running_tasks.client import TaskClientResultError
+from servicelib.utils import logged_gather
+
+from ....api.dependencies.database import get_base_repository
+from ....core.errors import NodeRightsAcquireError
+from ....core.settings import AppSettings, DynamicSidecarSettings
+from ....models.schemas.dynamic_services.scheduler import (
+    DockerContainerInspect,
+    DockerStatus,
+    SchedulerData,
+)
+from ...db.repositories import BaseRepository
+from ...director_v0 import DirectorV0Client
+from ...node_rights import NodeRightsManager, ResourceName
+from ..api_client import (
+    BaseClientHTTPError,
+    DynamicSidecarClient,
+    get_dynamic_sidecar_client,
+)
+from ..docker_api import (
+    get_projects_networks_containers,
+    remove_dynamic_sidecar_network,
+    remove_dynamic_sidecar_stack,
+    try_to_remove_network,
+)
+
+logger = logging.getLogger(__name__)
+
+
+# Used to ensure no more that X services per node pull or push data
+# Locking is applied when:
+# - study is being opened (state and outputs are pulled)
+# - study is being closed (state and outputs are saved)
+RESOURCE_STATE_AND_INPUTS: Final[ResourceName] = "state_and_inputs"
+
+
+@asynccontextmanager
+async def disabled_directory_watcher(
+    dynamic_sidecar_client: DynamicSidecarClient, dynamic_sidecar_endpoint: AnyHttpUrl
+) -> AsyncIterator[None]:
+    try:
+        # disable file system event watcher while writing
+        # to the outputs directory to avoid data being pushed
+        # via nodeports upon change
```

Comment:
maybe move these comments to the docstring? from the name of the functions I can get that it is disabled and re-enabled.